charts/oidc-proxy/templates/configmap.yaml

```diff
@@ -95,7 +114,23 @@ data:
             </Limit>
 
             ${AUTH_TYPE2}
-            ${AUTH_REQUIRE2}
+            <RequireAll>
+              <RequireAll>
+                ${AUTH_REQUIRE2}
+              </RequireAll>
+              <RequireAll>
+                # Note: we need to reference %{REMOTE_USER} to ensure that this expression is evaluated _after_ the authentication has happened
+                # %{REMOTE_USER} != %{REMOTE_USER} will never evaluate to true but will ensure that the expression is evaluated at the correct time
+                # An example expression can look like:
+                # Require expr "%{REMOTE_USER} != %{REMOTE_USER} || %{HTTP:OAUTH2_CLAIM_email} =~ /.gserviceaccount.com$/"
+                Require expr "%{REMOTE_USER} != %{REMOTE_USER} || ('${ENVIRONMENT}' in {'dev', 'alpha', 'perf', 'staging', 'prod'} && (${ALLOW_EXPRESSION}))"
```

Comment:
Curious if you set `$ALLOW_EXPRESSION` anywhere?

I was having trouble with this in some of the terra-helmfile proxies. The old proxy used the jq-based `Require claims_expr` which isn't possible in mod_oauth2. In the firecloud-develop proxies we are building the `Require expr` with go ctmpl logic. But for the helmfile configmap we can't use ctmpl on a rendered k8s secret (as far as I can tell). So kind of stuck on how to build this expression…

Reply:
Honestly, we don't even really need it.  I had it there more of as an example but realistically we depend on Sam to kick out users that don't meet the required clientid

Reply:
That's true -- a lot of the checks in the proxy are redundant if the service uses Sam. This probably goes for the LDAP check too..